### PR TITLE
ROCANA-9270: add forked postings format to Lucene's SPI

### PR DESCRIPTION
**Pull request by bradcupit**

### CODE REVIEW

File: src/test/java/com/rocana/lucene/codec/v1/TestRocanaLucene50PostingsFormat.java
Comment (matthew-dailey) [inline src/test/java/com/rocana/lucene/codec/v1/TestRocanaLucene50PostingsFormat.java]: Should this be exactly equal to `RocanaLucene50PostingsFormat.SHORT_NAME`?

Comment (bradcupit) [reply src/test/java/com/rocana/lucene/codec/v1/TestRocanaLucene50PostingsFormat.java]: true, we do have a public constant for that. I'll adjust the code

**Review by matthew-dailey — APPROVED** [review]
LGTM Brad!  Feel free to wait for more feedback before merging, but I think it looks good

File: src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat.java
Comment (bradcupit) [inline src/test/java/com/rocana/lucene/codec/v1/TestBlockPostingsFormat.java:76]: Interestingly adding our postings format to the SPI file made these tests (originally from Lucene's code) use our forked RocanaFieldReader and RocanaStats instead of the Lucene originals, which is excellent.

I had to make this change or the test failed with a ClassCastException.

File: src/main/resources/META-INF/services/org.apache.lucene.codecs.PostingsFormat
Comment (mpeterson0x0) [inline src/main/resources/META-INF/services/org.apache.lucene.codecs.PostingsFormat:24]: I know it's kind of late in the game here but the `v1` in the package name has been making me nervous for a while and now I know why - you've tied both the RocanaLucene50PostingsFormat and RocanaCodec (and probably the PerFieldsPostingsFormat) to be in "v1" together, but it is very likely that we will need to change them independently. I can see us needing to change/update the RocanaCodec, but not the `RocanaLucene50PostingsFormat`, but you would have to move both to a `v2` package even though one didn't change.

I think it is better to follow the Lucene convention of leaving them in an unversioned package and put the version in the name.  We already have the version in the name, (so yay).  My vote is to do like Lucene and put the RocanaCodec these in a "codecs" package with no version.

Comment (bradcupit) [reply src/main/resources/META-INF/services/org.apache.lucene.codecs.PostingsFormat:24]: I originally had it in the file name. @joey suggested moving it to the package name, and then in a PR review with you we talked about putting it back in the file name, so it would be in both.

But if I remove the version from the package name now then I've violated something @joey and I talked about and agreed to.

As to why we agreed to it, I'm sure Joey could defend it better than I could. I think the idea was we'd need to support two codecs in the future, both simultaneously, so we need the version in the package. I see that going two routes:

1. When we need a v2 codec we aren't also upgrading Lucene, in which case we're either our v2 codec is entirely custom (in which case there's no conflict and the separate package works out well), or it's based on some other Lucene classes we're not using yet (this also works out well for the v2 package), or we further customize our existing forked code, to the point where it's changing the on-disk representation enough that we need to increment the codec version to v2. That last one is the hairiest, but in that case I think we'd have to copy/paste the existing v1 code to the v2 package and then tweak it there. The reason I think we need this is because we'd need both the v1 and v2 codec at the same time.
2. We upgrade Lucene, which will be really messy, but in this case our v2 codec might be based off Lucene 7.0, for example, and maybe Lucene 7.0 has a different on disk format. In this case we can only have one version of Lucene on the classpath at once, and it would have to be the newer version. So we're relying on new Lucene still having the Lucene54 codec code. If so, then our v1 package can stay unmodified and our v2 package can contain newer versions of those files. Or, maybe some of those files aren't specific to a codec. In that case maybe we move them from v1 to a common package, or upgrade our forks to the latest Lucene version and then move them to a common package since they're shared between codecs.

I'm not sure how it's going to work out, but if we need a v2 codec in the future, we'll know more then. (Of course, you could make that same argument that we shouldn't put the version in the package until we know we need a v2 codec, and I can't argue with that).

**Pull request merged**
